src/api_client.rs

```diff
@@ -222,12 +222,15 @@ impl Client {
     }
 
     // v2 endpoint
-    pub async fn get_beacon_block(id: BlockId) -> Result<SignedBeaconBlock, Error> {
-        unimplemented!("")
+    pub async fn get_beacon_block(self, id: BlockId) -> Result<SignedBeaconBlock, Error> {
```

Comment:
let's take `self` by ref
```suggestion
    pub async fn get_beacon_block(&self, id: BlockId) -> Result<SignedBeaconBlock, Error> {
```